js/PivotReport.PivotTable.js

```diff
@@ -143,6 +172,35 @@ CRM.PivotReport.PivotTable = (function ($) {
     });
   };
 
+  /**
+   * Initializes the record filter form when avaible. This includes moving it
+   * inside the pivot table, initializing date pickers, and refreshing the pivot
+   * table when the form is submmited.
+   */
+  PivotTable.prototype.initRecordFilterForm = function () {
+    // skip if there are no record filter forms defined:
+    if (this.recordFilterForm.length === 0) {
+      return;
+    }
+
+    // moves the form so it's inside the pivot table:
```

Comment:
Not sure what and why we are moving stuff, could you explain?

Reply:
As explained in the call, the filter is defined in the report's template, as this filter is very specific to this report, and then the filter is moved from the template so it's inside the HTML generated by the pivot report plugin.